src/Github/Api/V3/CreateReleaseThroughApiCall.php

```diff
@@ -34,32 +40,40 @@ public function __invoke(
         $request = $this->messageFactory
             ->createRequest(
                 'POST',
-                self::API_ROOT . 'repos/' . $repository->owner() . '/' . $repository->name() . '/releases',
+                sprintf(self::API_URI, $repository->owner(), $repository->name()),
             )
             ->withAddedHeader('Content-Type', 'application/json')
             ->withAddedHeader('User-Agent', 'Ocramius\'s minimal API V3 client')
             ->withAddedHeader('Authorization', 'token ' . $this->apiToken);
 
-        $request
-            ->getBody()
-            ->write(Json\encode([
-                'tag_name' => $version->fullReleaseName(),
-                'name'     => $version->fullReleaseName(),
-                'body'     => $releaseNotes,
-            ]));
+        $request->getBody()->write(encode([
+            'tag_name' => $version->fullReleaseName(),
+            'name'     => $version->fullReleaseName(),
+            'body'     => $releaseNotes,
+        ]));
 
         $response = $this->client->sendRequest($request);
 
-        Psl\invariant($response->getStatusCode() >= 200 && $response->getStatusCode() <= 299, 'Failed to create release through GitHub API.');
+        $statusCode        = $response->getStatusCode();
+        $responseBody      = (string) $response->getBody();
+        $validResponseCode = $statusCode >= 200 && $statusCode <= 299;
+
+        if ($validResponseCode) {
+            $responseData = typed($responseBody, shape([
+                'html_url' => non_empty_string(),
+            ]));
 
-        $responseBody = $response
-            ->getBody()
-            ->__toString();
+            return new Uri($responseData['html_url']);
+        }
 
-        $responseData = Json\typed($responseBody, Type\shape([
-            'html_url' => Type\non_empty_string(),
+        $errorResponseData = typed($responseBody, shape([
+            'message' => non_empty_string(),
         ]));
 
-        return new Uri($responseData['html_url']);
+        invariant_violation(
+            'Failed to create release through GitHub API;' . PHP_EOL . 'Status code: %s' . PHP_EOL . 'Message: %s' . PHP_EOL,
```

Comment:
Unrelated: let's never use `PHP_EOL`, as it changes code execution depending on OS otherwise.